elastic_apm_falcon/__init__.py

```diff
@@ -18,8 +20,10 @@ def __init__(self, **config):
 
     def process_request(self, req: Request, resp: Response):
         if not self.client.should_ignore_url(req.path):
+            trace_parent = TraceParent.from_headers(req.headers)
+
             # begin APM transaction
-            self.client.begin_transaction("request")
+            self.client.begin_transaction("request", trace_parent=trace_parent)
```

Comment:
We have to be carefult here: Can `TraceParent.from_headers` raise any exceptions and/or can it return `None`? If yes, what happens if `self.client.begin_transaction` gets `trace_parent=None`?

Reply:
This is copy-pasted from the [Flask agent](https://github.com/elastic/apm-agent-python/blob/be6f4edd2c0ff11f565de759fa80e5b320039683/elasticapm/contrib/flask/__init__.py#L178-L179).

Falcon's `Request.headers` should always return a dictionary (see [here](https://github.com/falconry/falcon/blob/89e19dbabb747b889caf0b4ea5256ccb77d6351f/falcon/request.py#L889-L905)). Constructing `TraceParent.from_headers(req.headers)` with an empty dictionary returns `None`. The `trace_parent` argument is by default `None` (see [here](https://github.com/elastic/apm-agent-python/blob/be6f4edd2c0ff11f565de759fa80e5b320039683/elasticapm/base.py#L295)). So nothing bad should happen here.